### DISCUSSION TITLE
how to change the powershell version that quarto uses for post-render shell command?

### DISCUSSION THREAD

**Discussion by kompre** [quarto-dev/quarto-cli#8842] [Q&A] [answered]
### Description

I've just found out that quarto uses for post-render shell command a different version of powershell that I use as main. 

In any terminal window, the command `$PSVersionTable.PSVersion` return `7.4.1` (that's what I'm currently using), while the same command executed by quarto as post-render instruction

```yml
post-render:
  - powershell $PSVersionTable.PSVersion
```
will return `5.1`

This has created me much confusion because I couldn't understand why my script was failing: it was failing because I was triyng to use stuff not yet supported.

How can I tell quarto which powershell version to use?

**Comment by mcanouil** [comment]
That's controlled by your PATH I believe.

You should also be able to provide full path to binary in post render.

I am not on Windows and do not have PowerShell installed. Even-though I would have it, 5.1 is the version used internally by Windows.

CC @cderv

**Comment by cderv** [comment, marked as answer]
> In any terminal window, the command $PSVersionTable.PSVersion return 7.4.1 (that's what I'm currently using)

* `powershell` is the CLI command for powershell 5 version 
* `pwsh` is the CLI command for powershell 7+

See https://learn.microsoft.com/en-us/powershell/scripting/whats-new/differences-from-windows-powershell?view=powershell-7.4#renamed-powershellexe-to-pwshexe 

So here you are calling explicitly powershell 5 which is what you get. 

Quarto is just calling what ever command you pass using Deno.Run - it won't do anything else about shell configuration or detection.

**Comment by kompre** [comment]
> Quarto is just calling what ever command you pass using Deno.Run

this I was missing.

I fixed changing `powershell` to `pwsh -Command`. What a headache...

**Reply by cderv** [reply]
By doing 
````yaml 
post-render:
  - powershell $PSVersionTable.PSVersion
````

you asked quarto to run a command for you. The information about Deno.run being used does not change anything. 

For this to work 
- Command used must be in PATH 
- Right command need to be used - Quarto won't call a terminal by default for the user. 

Just wanted to clarify. 

Gald it works for you now !
